src/main/java/com/example/demo/services/BidService.java

```diff
@@ -0,0 +1,77 @@
+package com.example.demo.services;
+
+import java.sql.Timestamp;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import com.example.demo.entities.Bid;
+import com.example.demo.entities.Item;
+import com.example.demo.entities.User;
+import com.example.demo.exceptions.BidAmountLowException;
+import com.example.demo.exceptions.InvalidDataException;
+import com.example.demo.exceptions.NotFoundException;
+import com.example.demo.models.BidModel;
+import com.example.demo.repositories.BidsRepository;
+import com.example.demo.repositories.ItemsRepository;
+import com.example.demo.services.interfaces.IBidService;
+
+
+public class BidService implements IBidService{
+	
+	private BidsRepository bidsRepo;
+	private ItemsRepository itemsRepo;
+
+	public BidService(BidsRepository bidsRepo,ItemsRepository itemsRepo) {
+		this.bidsRepo=bidsRepo;
+		this.itemsRepo=itemsRepo;
+	}
+	
+	//tests if user bidder mismatch throws error
+	//if bid entity valid
+	//if checks max amount
+	//if checks base price
+	
+	@Override
+	public synchronized BidModel addBid(BidModel bidModel, User user) throws InvalidDataException,BidAmountLowException,NotFoundException {
+		if(bidModel.getBidder().getId()!=user.getId()) {
+			throw new InvalidDataException();
+		}
+		Bid bidEntity=Bid.fromModel(bidModel);
+		
+		OptionalDouble maxVal=bidsRepo.findByItemEquals(bidEntity.getItem()).stream().mapToDouble(x->x.getAmount().doubleValue()).max();
+		if(maxVal.isPresent()) {
+			if(maxVal.getAsDouble()>=bidEntity.getAmount().doubleValue()) {
+				throw new BidAmountLowException();
+			}
+		}else {
+			Optional<Item> item=itemsRepo.findById(bidEntity.getItem().getId());
+			if(item.isPresent()) {
+				if(item.get().getStartingprice().doubleValue()>bidEntity.getAmount().doubleValue()) {
+					throw new BidAmountLowException();
+				}
+			}else {
+				throw new NotFoundException();
+			}
+		}
+		Timestamp crr=new Timestamp(System.currentTimeMillis());
+		bidEntity.setTime(crr);
+		bidModel.setTime(crr);
+		return bidsRepo.save(bidEntity).toModel();
+	}
+	
+	//check item creation
+	//check to model
+
+	@Override
+	public Collection<BidModel> getBids(int itemId) {
+		Item item=new Item();
+		item.setId(itemId);
+		return bidsRepo.findByItemEquals(item).stream().map(x->x.toModel()).collect(Collectors.toList());
```

Comment:
Since we will have separate screen for bids, and here on item page, we should only show fixed ammount of bids

it would be better if we would not do this, we should fetch bids and item as separate requests from UI
we can have /item/:itemId/bids to get all bids and /item/:itemId/bids?limit=5 to fetch bids (already sorted) for product screen